src/policy/concrete.rs

```diff
@@ -278,6 +278,7 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
             )),
             _ => {
                 let (internal_key, policy) = self.clone().extract_key(unspendable_key)?;
+                policy.check_tapleaf()?;
```

Comment:
Should update the name of the function. This implies that we can checking something on tapleafs. Perhaps, `check_num_tapleaves`?
I would prefer to inline the `check_num_tapleaves` here, but up to you.

Reply:
Yes, the function name is a misnomer. Will change the name as suggested.